### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by michalvasko**

### CODE REVIEW

File: src/diff.c [hunk 1]
```diff
@@ -82,6 +82,21 @@ lyd_diff_add(const struct lyd_node *node, enum lyd_diff_op op, const char *orig_
 
     assert(diff);
 
+    /* replace leaf always needs orig-default and orig-value */
```
Comment (rkrejci) [inline src/diff.c:85]: Next time, I don't think that adding asserts is a FEATURE, it's just debugging not even changing anything in the Release build.

**Review by rkrejci — APPROVED** [review]

File: src/diff.c (hunk 1 above)
Comment (michalvasko) [reply src/diff.c:85]: Okay but these asserts have somewhat extended functionality of documenting (by code) what metadata exactly are expected in which case, that is why using MAINTENANCE or some similar tag did not seems appropriate to me.

Comment (rkrejci) [reply src/diff.c:85]: so then it's DOC, isn't it?

Comment (michalvasko) [reply src/diff.c:85]: Not sure, can DOC be something non-comment?

Comment (rkrejci) [reply src/diff.c:85]: can FEATURE be something that is not compiled? :)

Comment (michalvasko) [reply src/diff.c:85]: Like I said, I do not know what exactly I was supposed to use because none of the tags seem to be suitable.

**Pull request merged**